src/v/kafka/server/quota_manager.cc

```diff
@@ -38,6 +38,8 @@
 
 using namespace std::chrono_literals;
 
+const static unsigned quotas_shard = 0;
```

Comment:
Yeah that would be possible. But I don't expect either the memory overhead of the map or the CPU overhead of the inserts/background fibres to be large, so I think we should keep it simple. Having the global map distributed across the cores would make the background updates more complex (`gc()/update_quotas()` would need to work across multiple cores).

Reply:
> gc()/update_quotas() would need to work across multiple cores

But this should be trivial no? Each shard would simply kick off the GCs for their elements in the same way shard 0 currently does it for all.

Reply:
Yes, that's true. I was thinking about whether we need to hold a single lock while updating all the cores or if shard-local locks would suffice. I think the latter is sufficient because the global maps are independent so their updates/gc's/inserts can happen in any order. But I still think doing all the work on shard 0 makes this simpler.